### PR TITLE
Authentication Service

### LINKED ISSUE #106: Authentication Service
Create a service which can perform all the authentication functions which include:
- [ ] Google sign in
- [ ] Web endpoint communication
- [ ] Firebase passwordless login for NIT Rourkela webmail verification

### PR DESCRIPTION
**Pull request by ankank30**
Fixes #106 and #109

### CODE REVIEW

**Comment by Thesmader** [comment]
Do not merge yet. I will add a commit fixing the lint warnings.

**Review by Chinmay-KB — APPROVED** [review]

**Comment by ankank30** [comment]
Ready to merge

File: elaichi/lib/datamodels/data.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import 'package:elaichi/datamodels/auth_user.dart';
+
+///
```
Comment (Thesmader) [inline elaichi/lib/datamodels/data.dart:3]: @ankank30 please add something to this line to explain the purpose of the class.

File: elaichi/pubspec.yaml [hunk 2]
```diff
@@ -55,6 +63,9 @@ dev_dependencies:
   # Local database
   hive_generator: ^0.7.1
 
+  # graphql generator
+  gql_code_gen: ^0.1.5
```
Comment (Thesmader) [inline elaichi/pubspec.yaml:67]:
```suggestion

```

**Review by Thesmader — APPROVED** [review]

**Pull request merged**